### DISCUSSION TITLE
ProjectCapability   or   ProjectTypeGuids

### DISCUSSION THREAD

**Discussion by John0King** [CZEMacLeod/MSBuild.SDK.SystemWeb#64] [General]
talk before in #21 

after I looking to the  .targets and .props   I can not find `<ProjectCapability /> `   in  the orignal Sdk,  so I ask to ChatGPT , and it's answer is `<ProjectTypeGuids />`,
there are two  `ProjectTypeGuilds`  
eg. 

```xml
<ProjectTypeGuids>{349c5851-65df-11da-9384-00065b846f21};{fae04ec0-301f-11d3-bf4b-00c04f79efbc}</ProjectTypeGuids>
```

I'm wonder  will it better suite for  asp.net 4 web applications ?

**Comment by CZEMacLeod** [comment]
`ProjectTypeGuids` affect which features are enabled in a legacy/classic project, and are ignored by the CPS (the new project system which SDK projects derive from).
One of those `{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}` says it is c#
The other says it is ASP.NET 4.x

None of the VS extensions for legacy projects are enabled for the new CPS, so this doesn't have any effect.
The new replacement is the `ItemGroup` type `ProjectCapability` which allows you to add multiple flags to your project which then enable or adjust functionality in various VS extensions etc.
e.g. the line `<ProjectCapability Include="LegacyRazorEditor" />` means the LSP for .vbhtml and .cshtml files falls back to the original ASP.NET 4.x implementation and doesn't assume you are targeting aspnetcore.

This is why you also find a different project type id in the solution file for SDK projects vs. legacy projects.
